### ISSUE TITLE
The connection associated to this uuid is forbidden

### ISSUE THREAD

**Issue by jayb-g** [pcrockett/mollysocket-fly#11]
After doing `fly deploy`, when I open ssh console and run `mollysocket test uuid {uuid}` it gives:

```
 {uuid} is valid
The connection associated to this UUID is forbidden
```

Does it mean that uuid is blocked by signal? Is uuid related to signal account ?

Also when I run `mollysocket connection ping {uuid}` it shows .....Connection {
uuid: ...
device_id: 2,
...
...
forbidden: true,
...
}

Before fly deploy, it was false. I'm not sure if this is specific to my account or not, because as I mentioned previosuly in #6, it used to work perfectly after `fly deploy`

And hence notifications are not working after a delay of around 2 mins.

**Comment by pcrockett** [comment]
_sorry for the delay, haven't had time._

i looked around in mollysocket code and [found](https://github.com/mollyim/mollysocket/blob/3b9d52db02c97953115dca86ca5fe11045ec3558/src/server/connections.rs#L125) that "forbidden" here seems to be referring to an [http response code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403) from the signal service.

so in other words, it sounds like for some reason mollysocket is unable to authenticate with the signal service for this particular connection.

_that does NOT necessarily mean you've been blocked by signal._ it could just mean that some credentials expired, or something like that.

i think it would make sense to remove the connection by issuing the command `mollysocket connection remove <your_uuid>` and then reconfiguring the connection again.

i also recommend running `flyctl logs` while you're at it; you may get some useful information there while setting up your new connection.

**Comment by jayb-g** [comment]
Yes, the molly app still works while I keep getting forbidden true across multiple reintallations of mollysocket.

Haven't tried to troubleshoot further yet, just thought using signal would be better for the time. 

Will definitely try what you said and report back.

**Comment by pcrockett** [comment]
I am now experiencing this issue, so I can dig into it more. Will keep you posted. For what it's worth, my suggestion above didn't work for me.

**Comment by pcrockett** [comment]
Figured it out (for my case at least).

How many linked devices do you have? Signal allows a max of 3. Since MollySocket counts as a linked device, you can't use it if you already have 3 devices linked to your primary.

If you already have 3 linked devices, try these steps:

1. Unlink one of your devices
2. Remove the connection from MollySocket using `mollysocket connection remove ...`
3. Re-add the connection as described in the instructions
4. Restart (redeploy) the MollySocket server

I'm going to close this issue for now -- If that doesn't solve your problem, we can re-open it.

**Comment by jayb-g** [comment]
@pcrockett sounds reasonable. How to unlink devices if I have removed the fly.io instance already ? How do I know if my account already has x number of linked devices ?

**Comment by pcrockett** [comment]
I just now added a section to the troubleshooting doc: https://github.com/pcrockett/mollysocket-fly/blob/main/doc/TROUBLESHOOTING.md#you-have-too-many-linked-devices

In the Molly Android app, in settings, it should show you a list of linked devices. Tap on one of them to unlink. More details in the troubleshooting document.

**Comment by jayb-g** [comment]
@pcrockett seems to be working fine. And mice improvements to the documentation including automatic updates. 

One thing I wanted to ask, if you recommend websockets vs json stream over http for ntfy ? How are they different in terms of battery usage and reliability ?

**Comment by jayb-g** [comment]
And after 

> Back on your terminal, while you're in your cloned repository, run flyctl tokens create deploy. Then populate the FLY_DEPLOY_TOKEN repository secret with the output of that command.

in the _HOWTO.md_, add instructions to enable Deploy under Actions tab since its disabled by default.

**Comment by pcrockett** [comment]
@jayb-g thanks.

> recommend websockets vs json stream over http for ntfy ? How are they different in terms of battery usage and reliability ?

i just do the default, haven't had any trouble with it.

> in the HOWTO.md, add instructions to enable Deploy under Actions tab since its disabled by default.

🤔 ok thanks for the heads up. something i'll have to tackle later though.

glad it's working! if you need anything else, feel free to create a new issue.
